src/xscen/testing.py

```diff
@@ -242,3 +249,185 @@ def fake_data(
+
+    if not changes_file.exists():
+        raise FileNotFoundError("Changes file not found in xscen file tree.")
+
+    with Path(changes_file).open(encoding="utf-8") as f:
+        changes = f.read()
+
+    if style == "rst":
+        hyperlink_replacements = {
+            r":issue:`([0-9]+)`": r"`GH/\1 <https://github.com/Ouranosinc/xscen/issues/\1>`_",
+            r":pull:`([0-9]+)`": r"`PR/\1 <https://github.com/Ouranosinc/xscen/pull/\>`_",
+            r":user:`([a-zA-Z0-9_.-]+)`": r"`@\1 <https://github.com/\1>`_",
+        }
+    elif style == "md":
+        hyperlink_replacements = {
+            r":issue:`([0-9]+)`": r"[GH/\1](https://github.com/Ouranosinc/xscen/issues/\1)",
+            r":pull:`([0-9]+)`": r"[PR/\1](https://github.com/Ouranosinc/xscen/pull/\1)",
+            r":user:`([a-zA-Z0-9_.-]+)`": r"[@\1](https://github.com/\1)",
+        }
+    else:
+        raise NotImplementedError()
+
+    for search, replacement in hyperlink_replacements.items():
+        changes = re.sub(search, replacement, changes)
+
+    if latest:
+        changes_split = changes.split("\n\nv0.")
+        changes = changes_split[0] + "\n\nv0." + changes_split[1]
+
+    if style == "md":
+        changes = changes.replace("=========\nChangelog\n=========", "# Changelog")
+
+        titles = {r"\n(.*?)\n([\-]{1,})": "-", r"\n(.*?)\n([\^]{1,})": "^"}
+        for title_expression, level in titles.items():
+            found = re.findall(title_expression, changes)
+            for grouping in found:
+                fixed_grouping = (
+                    str(grouping[0]).replace("(", r"\(").replace(")", r"\)")
+                )
+                search = rf"({fixed_grouping})\n([\{level}]{'{' + str(len(grouping[1])) + '}'})"
+                replacement = f"{'##' if level == '-' else '###'} {grouping[0]}"
+                changes = re.sub(search, replacement, changes)
+
+        link_expressions = r"[\`]{1}([\w\s]+)\s<(.+)>`\_"
+        found = re.findall(link_expressions, changes)
+        for grouping in found:
+            search = rf"`{grouping[0]} <.+>`\_"
+            replacement = f"[{str(grouping[0]).strip()}]({grouping[1]})"
+            changes = re.sub(search, replacement, changes)
+
+    if not file:
+        return changes
+    if isinstance(file, Path | os.PathLike):
+        file = Path(file).open("w")
+    print(changes, file=file)
+
+
+def show_versions(
+    file: os.PathLike | StringIO | TextIO | None = None,
+    deps: list | None = None,
+) -> str | None:
+    """Print the versions of xscen and its dependencies.
+
+    Parameters
+    ----------
+    file : {os.PathLike, StringIO, TextIO}, optional
+        If provided, prints to the given file-like object. Otherwise, returns a string.
+    deps : list, optional
+        A list of dependencies to gather and print version information from. Otherwise, prints `xscen` dependencies.
+
+    Returns
+    -------
+    str or None
+    """
+    if deps is None:
+        deps = [
```

Comment:
I'm not sure we will remember to modify this when we change deps. Is there a way to pull it from somewhere without explicetely naming them all?

Reply:
I've found a way to do that (and by "I", I mean Copilot wrote it all).